fix: to #157015 fix view label command localized

yiliang114 opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->
Fixes #157015

TylerLeonhardt commented:
Awesome!

Looks like you do have a couple of issues still:
```
[tsec-compile-check      ] src/vs/workbench/contrib/scm/browser/scm.contribution.ts(115,2): error TS2322: Type 'string' is not assignable to type 'ILocalizedString'.
[tsec-compile-check      ] src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts(181,32): error TS2352: Conversion of type '{ id: string; name: string; ctorDescriptor: SyncDescriptor<TreeViewPane>; when: RawContextKey<boolean>; canToggleVisibility: true; canMoveView: false; treeView: TreeView; collapsed: false; hideByDefault: false; }' to type 'ITreeViewDescriptor' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
[tsec-compile-check      ]   Types of property 'name' are incompatible.
[tsec-compile-check      ]     Type 'string' is not comparable to type 'ILocalizedString'.
[tsec-compile-check      ] 
[tsec-compile-check      ] 
[tsec-compile-check      ] Found 2 errors.
[tsec-compile-check      ] error Command failed with exit code 1.
```
From the CI

yiliang114 commented on src/vs/workbench/contrib/quickaccess/browser/viewQuickAccess.ts:
```diff
@@ -104,7 +104,7 @@ export class ViewQuickAccessProvider extends PickerQuickAccessProvider<IViewQuic
 				if (this.contextKeyService.contextMatchesRules(view.when)) {
 					result.push({
-						label: view.name,
+						label: view.name.original,
```
Use original here to display original text when other places are localized

TylerLeonhardt replied:
Not sure I quite understand what you mean here. Why original? (also could you add a comment saying why)

yiliang114 commented:
> Awesome!
> 
> Looks like you do have a couple of issues still:
> 
> ```
> [tsec-compile-check      ] src/vs/workbench/contrib/scm/browser/scm.contribution.ts(115,2): error TS2322: Type 'string' is not assignable to type 'ILocalizedString'.
> [tsec-compile-check      ] src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts(181,32): error TS2352: Conversion of type '{ id: string; name: string; ctorDescriptor: SyncDescriptor<TreeViewPane>; when: RawContextKey<boolean>; canToggleVisibility: true; canMoveView: false; treeView: TreeView; collapsed: false; hideByDefault: false; }' to type 'ITreeViewDescriptor' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
> [tsec-compile-check      ]   Types of property 'name' are incompatible.
> [tsec-compile-check      ]     Type 'string' is not comparable to type 'ILocalizedString'.
> [tsec-compile-check      ] 
> [tsec-compile-check      ] 
> [tsec-compile-check      ] Found 2 errors.
> [tsec-compile-check      ] error Command failed with exit code 1.
> ```
> 
> From the CI

Thanks!

Now it is reasonable to say that all the processing has been completed.
![image](https://github.com/microsoft/vscode/assets/11473889/356405a4-1a9a-4820-bfe9-7191ec2d5df7)

(I think I should have handled all the compilation errors before. When I update the main branch code, some code may be restored due to automatic merging)

TylerLeonhardt commented on src/vs/workbench/contrib/editSessions/browser/editSessionsViews.ts:
```diff
@@ -47,7 +47,7 @@ export class EditSessionsDataViews extends Disposable {
 		viewsRegistry.registerViews([<ITreeViewDescriptor>{
 			id: viewId,
-			name,
+			name: { value: name, original: name },
```
This one should be updated by either making `EDIT_SESSIONS_TITLE` an `ILocalizedString` or have this be:

```ts
name: { value: name, original: '......' },
```

TylerLeonhardt commented on src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts:
```diff
@@ -180,7 +180,7 @@ export class UserDataSyncDataViews extends Disposable {
 		viewsRegistry.registerViews([<ITreeViewDescriptor>{
 			id,
-			name,
+			name: { value: name, original: name },
```
this is using name for both.

TylerLeonhardt commented on src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts:
```diff
@@ -992,7 +992,7 @@ export class UserDataProfileImportExportService extends Disposable implements IU
 		const descriptor: ITreeViewDescriptor = {
 			id,
-			name,
+			name: { value: name, original: name },
```
this is using name for both.

yiliang114 replied:
The name here is passed in through the `showProfilePreviewView` function and looks like a user-defined profile name. Does this value determine that name is different from original ?

TylerLeonhardt reviewed (commented):
This is super close! Just a few comments

TylerLeonhardt commented on src/vs/workbench/browser/parts/views/viewsService.ts:
```diff
@@ -485,7 +485,7 @@ export class ViewsService extends Disposable implements IViewsService {
+				const title = localize({ key: 'focus view', comment: ['{0} indicates the name of the view to be focused.'] }, "Focus on {0} View", viewDescriptor.name.value);
 				super({
 					id: viewDescriptor.focusCommand ? viewDescriptor.focusCommand.id : `${viewDescriptor.id}.focus`,
 					title: { original: `Focus on ${viewDescriptor.name} View`, value: title },
```
I think you need to change this to `viewDescriptor.name.original` no? This is actually the fix you're fixing. You tested this fix, right?

yiliang114 replied:
> I think you need to change this to `viewDescriptor.name.original` right? Otherwise you will see `[Object object]` You may want to spot check the other references to make sure you are using the correct value.

Yes, yes, I also found this problem and have already disposed of it. Now it behaves like this in the Chinese environment:

![image](https://github.com/microsoft/vscode/assets/11473889/13dcef28-2108-4138-83e9-5aed021ac090)

TylerLeonhardt commented:
FYI we are currently in the middle of a release right now, so I want to hold off on getting this in til next week. That way we can let it sit in Insiders for a bit to make sure we don't cause any regressions due to the number of files changed. I think the PR is ready, but I'll approve and merge early next week.

yiliang114 commented on src/vs/workbench/services/views/test/browser/viewDescriptorService.test.ts:
```diff
@@ -60,19 +60,19 @@ suite('ViewDescriptorService', () => {
 			{
```
Thanks for the help!

TylerLeonhardt commented:
@yiliang114 I hope you don't hate me for this but I just added a new thing that will make this PR much smaller and have less duplicate strings.

I have added a `localize2` function [in this PR](https://github.com/microsoft/vscode/pull/194750) that allows us to replace all of these:
```ts
{ value: nls.localize({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console"), original: 'Debug Console' }
```
with a much simpler:
```ts
nls.localize2({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console")
```

this function returns an `ILocalizedString` and handles the original string correctly.

Can you move over to using that so the code is easier to maintain and we don't need to have two of the same string next to each other?

yiliang114 commented:
> @yiliang114 I hope you don't hate me for this but I just added a new thing that will make this PR much smaller and have less duplicate strings.
> 
> I have added a `localize2` function [in this PR](https://github.com/microsoft/vscode/pull/194750) that allows us to replace all of these:
> 
> ```ts
> { value: nls.localize({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console"), original: 'Debug Console' }
> ```
> 
> with a much simpler:
> 
> ```ts
> nls.localize2({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console")
> ```
> 
> this function returns an `ILocalizedString` and handles the original string correctly.
> 
> Can you move over to using that so the code is easier to maintain and we don't need to have two of the same string next to each other?
> 
> I think this won't apply to _everything_ in your PR but probably about 80-90% if the cases.

Haha, it looks better this way. Let me try again.

yiliang114 commented:
Please help me review the code again when you are free @TylerLeonhardt , I did not encountered an error  when I compiled locally.

![image](https://github.com/microsoft/vscode/assets/11473889/791472c3-66d4-4bda-a52d-bd5f9778da4b)

TylerLeonhardt reviewed (approved):
LGTM thanks for taking this on!

roblourens commented on src/vs/workbench/services/views/test/browser/viewContainerModel.test.ts:
```diff
@@ -63,13 +64,13 @@ suite('ViewContainerModel', () => {
 
 	test('empty model', function () {
-		container = ViewContainerRegistry.registerViewContainer({ id: 'test', title: { value: 'test', original: 'test' }, ctorDescriptor: new SyncDescriptor(<any>{}) }, ViewContainerLocation.Sidebar);
+		container = ViewContainerRegistry.registerViewContainer({ id: 'test', title: nls.localize2('test', 'test'), ctorDescriptor: new SyncDescriptor(<any>{}) }, ViewContainerLocation.Sidebar);
```
These will be excluded from the actual localization process, right?

TylerLeonhardt replied:
Yep!
